data/src/test/java/com/ragdroid/data/MainRepositorySpec.kt

```diff
@@ -0,0 +1,83 @@
+package com.ragdroid.data
+
+import com.ragdroid.api.MarvelApi
+import com.ragdroid.data.base.Helpers
+import com.ragdroid.data.entity.AppConfig
+import com.ragdroid.data.entity.CharacterMapper
+import com.ragdroid.data.entity.CharacterMarvel
+import io.reactivex.Single
+import io.reactivex.observers.TestObserver
+import org.amshove.kluent.mock
+import org.jetbrains.spek.api.Spek
+import org.jetbrains.spek.api.dsl.given
+import org.jetbrains.spek.api.dsl.it
+import org.jetbrains.spek.api.dsl.on
+import org.jetbrains.spek.api.dsl.xit
+import org.mockito.Mockito.*
+
+/**
+ * Created by garimajain on 09/12/17.
+ */
+
+object MainRepositorySpec: Spek({
+    given("a repository") {
+        var mockApi = mock(MarvelApi::class)
+        var mapper = mock(CharacterMapper::class)
+        var config = mock(AppConfig::class)
+        var helpers = mock(Helpers::class)
+        var repository = MainRepositoryImpl(mockApi, mapper, config, helpers)
+
+        beforeEachTest {
+            mockApi = mock(MarvelApi::class)
+            mapper = mock(CharacterMapper::class)
+            config = mock(AppConfig::class)
+            helpers = mock(Helpers::class)
+
+            `when`(config.publicKey).thenReturn("publicKEY")
```

Comment:
Hard keywords are a bad programming experience. Alternatively you can use `typealias` or simple import a line like
```
import org.mockito.Mockito.`when` as whenever
```
And use `whenever` instead of `'when'`.

Reply:
Cool, I will use `whenever` `wherever`